### PR TITLE
Start a React-ion of the stream page … sigh

### PR DESCRIPTION
**Pull request by matthew-andrews**
Justification:-

- Stream pages (e.g. FastFT) will have more advanced auto-reloading functionality which will not be user triggered, so accessibility type things like keeping selected text is more important than before.

Note the React version of the stream page is just a simple list for now…

### CODE REVIEW

File: components/stream-items/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		const items = context.items.map(item => {
+				return <li>{item.headline}</li>;
+			});
+
+		return (
+			<section role="section">
+				<div className="o-grid-row">
+					<div data-o-grid-colspan="full-width">
+						<h2 className="stream__title">{title}</h2>
+						<ol className="stream__list js-stream-list" data-trackable="{context.trackable}" id="{odId}">{items}</ol>
```
Comment (ironsidevsquincy) [inline components/stream-items/index.js]: minor: no need for quotes around attribute values if the values in `{...}`

`data-trackable="{context.trackable}" => data-trackable={context.trackable}`

File: components/stream-items/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		const sectionClass = (context.isLive ? 'live-stream js-live-stream' : '') + ' ' + (context.isHubPage ? 'topic-stream' : '');
+		const odId = context.isLive ? 'live-stream-items' : '';
+
+		const title = context.isSpecialReport
+			? `There are ${context.indexCount} articles in this special report`
+			: `Latest ${context.titlePreposition} ${context.title}`;
+
+		const relatedVideo = this.props.flags.relatedVideoStream
+			? <ol class="related-video-stream o-grid-row" data-trackable="related-video-stream"></ol>
+			: '';
```
Comment (ironsidevsquincy) [inline components/stream-items/index.js]: *think* it's better to use `null`, rather than empty string

File: components/stream-items/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, {Component} from 'react';
+
+export default class Layout extends Component {
+
+	render () {
+		const context = this.props.context;
+
+		const sectionClass = (context.isLive ? 'live-stream js-live-stream' : '') + ' ' + (context.isHubPage ? 'topic-stream' : '');
```
Comment (ironsidevsquincy) [inline components/stream-items/index.js]: where's this used?

File: server/app.js [hunk 4]
```diff
@@ -1,43 +1,13 @@
...
+		'es6.arrowFunctions',
+		'es6.destructuring',
+		'es6.modules',
+		'es6.parameters',
+		'es6.spread',
+
+		'react',
+		'strict'
+	]
+});
```
Comment (ironsidevsquincy) [inline server/app.js]: is this needed for react?

**Comment by ironsidevsquincy** [comment]
Worth a test, :+1: 

One thing I'm struggling with (though quite possibly due to my limited React knowledge) is how to change the HTML outside of react, i.e. we use o-date, n-video, etc, to update on the client, but React has a minor freak-out when we do this, along the lines of 'something else has update the dom which wasn't me, so I'm reverting it back to what I know'

File: server/app.js (hunk 4 above)
Comment (matthew-andrews) [inline server/app.js]: just copy and pasted from front page…… definitely need babel for JSX

File: components/stream-items/index.js (hunk 2 above)
Comment (matthew-andrews) [inline components/stream-items/index.js]: ok

File: components/stream-items/index.js (hunk 3 above)
Comment (matthew-andrews) [inline components/stream-items/index.js]: fixed now

File: server/app.js (hunk 4 above)
Comment (ironsidevsquincy) [inline server/app.js]: have a card to test the performance of using babel register. guess it's ok to add to streams for now

Comment (matthew-andrews) [inline server/app.js]: let me see if i can eliminate it

**Comment by matthew-andrews** [comment]
I've reduce the babel-usage down to just JSX

**Pull request merged**